project-addons/sale_custom/wizard/sale_report_xls_wizard.py

```diff
@@ -0,0 +1,210 @@
+            move_line_ids = move_line_obj.search_read([('account_id.type', '=', 'receivable'),
+                                                      ('reconcile_id', '=', False),
+                                                      ('payment_term_id', 'not ilike', 'Pago inmediato'),
+                                                      ('stored_invoice_id', '!=', False)], ['maturity_residual'])
+            autocartera = 0.0
+            for move_line in move_line_ids:
+                autocartera += move_line['maturity_residual']
+            return autocartera
+
+    @api.multi
+    def _get_inventory_value(self, date_search):
+        """
+        Calculate stock inventory value at date_search.
+        :param date_search:
+        :return Float:
+        """
+        for sale_report in self:
+            #name = False
+            #attr = False
+            #ctx = dict(self._context)
+            #ctx.update({
+            #    'history_date': history_date
+            #})
+            stock_history_obj = sale_report.env['stock.history']
+            #res = stock_history_obj._get_inventory_value(False, False, context=ctx)
+            product_tmpl_obj = sale_report.env["product.template"]
+            product_obj = sale_report.env["product.product"]
+            res = 0.0
+            stock_history_data = stock_history_obj.search_read([('date', '<=', date_search),
+                                                                ('location_id', 'not in', (23, 30, 33))],
+                                                                ['quantity', 'price_unit_on_quant', 'product_id', 'company_id'])
+
+            for line in stock_history_data:
+                if line['product_id']:
+                    product_data = product_obj.search_read([('id', '=', line['product_id'][0])], ['product_tmpl_id', 'cost_method'])
+                    if product_data:
+                        product = product_data[0]
+                        product_tmpl_data = product_tmpl_obj.search_read([('id', '=', product['product_tmpl_id'][0])], ['id'])
+                        if product_tmpl_data:
+                            product_tmpl = product_tmpl_data[0]
+                            if product['cost_method'] == 'real':
+                                res += float(line['quantity'] * line['price_unit_on_quant'])
+                            else:
+                                res += float(line['quantity'] *\
+                                             product_tmpl_obj.get_history_price(product_tmpl['id'],
+                                                                                line['company_id'][0], date=date_search,
+                                                                                context=sale_report.env.context))
+
+            return res
+
+    @api.multi
+    def _select(self):
+        this_str = """ 
+            SELECT date_confirm::date, sum(price_total) as daily_sales,
```

Comment:
¿sumáis las dos compañías adredre no?

Reply:
Si lo hicieras con un search si filtraría por la compañía del usuario que lo ejecuta porque eso se hace con una regla de seguridadad que tiene sale_report, pero al hacerlo con una query no se va a filtrar.

Reply:
Hmmm entiendo, pero entonces el resultado no me deberia de salir igual en el informe que los datos que me da Odoo desde la interfaz y si que lo hace

Reply:
Ok, eso es porque en la otra compañías no hay ventas, sólo facturas, por lo que no suma nada